### PR TITLE
created view model for UserProfile Activity

### LINKED ISSUE #2122: Create ViewModel for UserProfileActivity
**Actual Behaviour**

There is no ViewModel in UserProfileActivity

**Expected Behaviour**

There should be one.

**Would you like to work on the issue?**

Yes!

### PR DESCRIPTION
**Pull request by simarsingh24**
Fixes #2122 

Changes: added view model for userProfileActivity

### CODE REVIEW

File: android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java [hunk 1]
```diff
@@ -122,18 +120,18 @@ protected void onDestroy() {
         super.onDestroy();
         if (swipeRefreshLayout != null)
             swipeRefreshLayout.setOnRefreshListener(null);
+    }
+    private void getUserFromRealm() {
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java]: I think naming it getUser would be sufficient.

File: android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java [hunk 2]
```diff
@@ -122,18 +120,18 @@ protected void onDestroy() {
...
+        userProfileActivityViewModel.getUser().observe(this,user -> {
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java]: Add a space after the ```,```

File: android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java [hunk 3]
```diff
@@ -143,27 +141,20 @@ private void loadUser(boolean loadFromNetwork) throws JSONException {
...
+                        case UserProfileActivityViewModel.RETRIEVE_UNSUCCESSFUL:
                             stopRefreshing();
-                        },
-                        () -> {
+                            break;
+                        case UserProfileActivityViewModel.RETRIEVE_COMPLETE:
                             stopRefreshing();
-                            Timber.d("User data loaded from network");
-                        });
+                            break;
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java]: Add a default case for the switch statement. If you don't have anything to implement in that case, just add a comment "No implementation"

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    private MutableLiveData<User> userLiveData;
+    private RealmDataRepository realmRepo;
+    private Disposable loadUserDisposable;
+    private MutableLiveData<Integer> loadUserFromNetworkResponse;
+    public static final int RETRIEVE_SUCCESSFUL = 1;
+    public static final int RETRIEVE_UNSUCCESSFUL = 2;
+    public static final int RETRIEVE_COMPLETE = 3;
+
+    public UserProfileActivityViewModel() {
+        realmRepo=RealmDataRepository.getDefaultInstance();
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Add spaces before and after ```=```

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+public class UserProfileActivityViewModel extends ViewModel {
+
+    private User user;
+    private MutableLiveData<User> userLiveData;
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Use realm LiveData wrapper.

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    private RealmDataRepository realmRepo;
+    private Disposable loadUserDisposable;
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Use a final CompositeDisposable here. That would mean you won't need the null check in the ```onCleared```.

**Review by dr0pdb — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 7]
```diff
@@ -2,74 +2,77 @@
...
             user = realmRepo.getUser();
-            user.addChangeListener(realmModel -> {
-                userLiveData.setValue(user);
-                Timber.d("User data loaded from disk");
-            });
+            if(user == null){
+                user = new User();
+            }
+            LiveRealmDataObject<User> liveRealmDataObject = RealmDataRepository.asLiveDataForObject(realmRepo.getUser());
+            userLiveData = Transformations.map(liveRealmDataObject, input -> input);
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Why do we need this?

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 8]
```diff
@@ -2,74 +2,77 @@
...
         if(userLiveData == null){
-            userLiveData = new MutableLiveData<>();
             user = realmRepo.getUser();
-            user.addChangeListener(realmModel -> {
-                userLiveData.setValue(user);
-                Timber.d("User data loaded from disk");
-            });
+            if(user == null){
+                user = new User();
+            }
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Why is this needed

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 9]
```diff
@@ -2,74 +2,77 @@
...
     }
 
     public void clearUserDataFromRealm() {
         realmRepo.clearUserData();
     }
-    
     public LiveData<User> getUser() {
         if(userLiveData == null){
-            userLiveData = new MutableLiveData<>();
             user = realmRepo.getUser();
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Why is this needed

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+    public void clearUserDataFromRealm() {
+        realmRepo.clearUserData();
+    }
+    public LiveData<User> getUser() {
+        if(userLiveData == null){
+            user = realmRepo.getUser();
+            if(user == null){
+                user = new User();
+            }
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Why do we need this

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+        realmRepo = RealmDataRepository.getDefaultInstance();
+        compositeDisposable = new CompositeDisposable();
+    }
+
+    public void clearUserDataFromRealm() {
+        realmRepo.clearUserData();
+    }
+    public LiveData<User> getUser() {
+        if(userLiveData == null){
+            user = realmRepo.getUser();
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Why do we need this

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+                                })
+        );
+        return loadUserFromNetworkResponse;
+    }
+
+    @Override
+    protected void onCleared() {
+       compositeDisposable.dispose();
+        super.onCleared();
+    }
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Indentation is wrong

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java (hunk 11 above)
Comment (simarsingh24) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: no its not needed, i will make the changes right away :)

File: android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java [hunk 13]
```diff
@@ -68,9 +65,10 @@ protected void onCreate(Bundle savedInstanceState) {
 
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
 
+        userProfileActivityViewModel = ViewModelProviders.of(this).get(UserProfileActivityViewModel.class);
         if (!AuthUtil.isUserLoggedIn()) {
             //Delete User data from realm
-            RealmDataRepository.getDefaultInstance().clearUserData();
+            userProfileActivityViewModel.clearUserDataFromRealm();
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java]: Don't use implementation specific names. We will migrate from Realm soon. So it should be agnostic

File: android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java [hunk 14]
```diff
@@ -122,18 +120,18 @@ protected void onDestroy() {
         super.onDestroy();
         if (swipeRefreshLayout != null)
             swipeRefreshLayout.setOnRefreshListener(null);
+    }
+    private void getUser() {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java]: Add a space above and rename to subscribeUser

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+
+    public void clearUserDataFromRealm() {
+        realmRepo.clearUserData();
+    }
+    public LiveData<User> getUser() {
+        LiveRealmDataObject<User> liveRealmDataObject = RealmDataRepository.asLiveDataForObject(realmRepo.getUser());
+        Timber.d("User data loaded from disk");
+        return  liveRealmDataObject;
+    }
+    public LiveData<Integer> loadUserFromNetwork(long id) {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Add new line after each method

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+    public UserProfileActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+        compositeDisposable = new CompositeDisposable();
+    }
+
+    public void clearUserDataFromRealm() {
+        realmRepo.clearUserData();
+    }
+    public LiveData<User> getUser() {
+        LiveRealmDataObject<User> liveRealmDataObject = RealmDataRepository.asLiveDataForObject(realmRepo.getUser());
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: You only need to get it once

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+                                },
+                                throwable -> {
+                                    loadUserFromNetworkResponse.setValue(RETRIEVE_UNSUCCESSFUL);
+                                    Timber.d(throwable.getMessage() + " Error getting data from network");
+                                },
+                                () -> {
+                                    loadUserFromNetworkResponse.setValue(RETRIEVE_COMPLETE);
+                                    Timber.d("User data loaded from network");
+                                })
+        );
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: There's too much nesting. Move api code above and save the disposable in a variable and add in composite disposable below

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java (hunk 14 above)
Comment (iamareebjamal) [reply android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java]: Unchanged

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+import org.fossasia.openevent.dbutils.LiveRealmDataObject;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+public class UserProfileActivityViewModel extends ViewModel {
+    private CompositeDisposable compositeDisposable;
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Make it final

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+                        Timber.d("User data loaded from network");
+                });
+
+        compositeDisposable.add(disposable);
+        return loadUserFromNetworkResponse;
+    }
+
+    @Override
+    protected void onCleared() {
+        compositeDisposable.dispose();
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java:71]: Add a check whether compositeDisposable has already been disposed or not.

**Review by dr0pdb — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java (hunk 19 above)
Comment (simarsingh24) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java:71]: checking if it is disposed to dispose is doesnt make sense

Comment (simarsingh24) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java:71]: also in dispose() function if the composite Disposable is already disposed it simply returns

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java (hunk 18 above)
Comment (simarsingh24) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: why should it be final? we need to addDisposables to this variable.

Comment (dr0pdb) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Making it final won't stop you from adding disposables to it. Disposables are it's fields while making it final would stop you from creating a new instance of it.

Comment (simarsingh24) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: okay I will do it . thanks :)

File: android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java (hunk 14 above)
Comment (simarsingh24) [reply android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java]: Oh i refactored other function inside ViewModel ! I will make the changes right away

**Review by iamareebjamal — DISMISSED** [review]

**Comment by iamareebjamal** [comment]
@simarsingh24 Please squash the commits

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+            Timber.d("User data loaded from disk");
+        }
+        return  liveRealmDataObject;
+    }
+
+    public LiveData<Integer> loadUserFromNetwork(long id) {
+        if(loadUserFromNetworkResponse == null){
+            loadUserFromNetworkResponse = new MutableLiveData<>();
+        }
+        Disposable disposable = APIClient.getOpenEventAPI()
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: No need to create this intermediate variable ```disposable```. Just add it straight away to the CompositeDisposable.

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+                .getUser(id)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(user -> {
+                    realmRepo.saveUser(user).subscribe();
+                    loadUserFromNetworkResponse.setValue(RETRIEVE_SUCCESSFUL);
+                    Timber.d("User data saved in database");
+                    },
+                        throwable -> {
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: indentation is off here in the onError.

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+    public LiveData<Integer> loadUserFromNetwork(long id) {
+        if(loadUserFromNetworkResponse == null){
+            loadUserFromNetworkResponse = new MutableLiveData<>();
+        }
+        Disposable disposable = APIClient.getOpenEventAPI()
+                .getUser(id)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(user -> {
+                    realmRepo.saveUser(user).subscribe();
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: You can chain this call using map completeable or equivalent.

**Review by dr0pdb — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java (hunk 20 above)
Comment (iamareebjamal) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: If indentation is changing a lot, this is the preferred way

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java (hunk 21 above)
Comment (simarsingh24) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: that's not off, IDE automatically takes you there when you hit enter, that's the way Android Studio recommend it that means. I believe keep it like that would be good

Comment (iamareebjamal) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: It is off `},` should be either 1 indent less than previous indentation or previous should be 1 more

Secondly, throwable should be on previous line and its block should be one indent less

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java (hunk 22 above)
Comment (simarsingh24) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: yeah sure but then where would I add this part of the code:
 `() -> {
                                            loadUserFromNetworkResponse.setValue(RETRIEVE_COMPLETE);
                                            Timber.d("User data loaded from network");
                                    });`

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java (hunk 21 above)
Comment (simarsingh24) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: okay i will make the changes 👍

**Comment by simarsingh24** [comment]
@srv-twry  @iamareebjamal  please review the changes

File: android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java [hunk 23]
```diff
@@ -143,27 +142,20 @@ private void loadUser(boolean loadFromNetwork) throws JSONException {
...
-                                    .getDefaultInstance()
-                                    .saveUser(user)
-                                    .subscribe();
-                            this.user = user;
-                            showUserInfo(user);
-                            Timber.d("User data saved in database");
-                        },
-                        throwable -> {
-                            Timber.d(throwable.getMessage() + " Error getting data from network");
+        userProfileActivityViewModel.loadUserFromNetwork(JWTUtils.getIdentity(AuthUtil.getAuthorization()))
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/auth/UserProfileActivity.java]: Pass `AuthUtil.getAuthorization()`

Calculate JWT in viewmodel

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 24]
```diff
@@ -0,0 +1,71 @@
...
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+public class UserProfileActivityViewModel extends ViewModel {
+    private final CompositeDisposable compositeDisposable;
+    private RealmDataRepository realmRepo;
+    private MutableLiveData<Integer> loadUserFromNetworkResponse;
+    public static final int RETRIEVE_SUCCESSFUL = 1;
+    public static final int RETRIEVE_UNSUCCESSFUL = 2;
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Hoist constant to top

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 25]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    public void eraseUserData() {
+        realmRepo.clearUserData();
+    }
+
+    public LiveData<User> getUser() {
+        if(liveRealmDataObject == null) {
+            liveRealmDataObject = RealmDataRepository.asLiveDataForObject(realmRepo.getUser());
+            Timber.d("User data loaded from disk");
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Remove log. User isn't loaded till observed

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 26]
```diff
@@ -0,0 +1,71 @@
...
+                .getUser(id)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .flatMapCompletable(user -> RealmDataRepository
+                        .getDefaultInstance()
+                        .saveUser(user))
+                .subscribe(() -> {
+                    loadUserFromNetworkResponse.setValue(RETRIEVE_SUCCESSFUL);
+                    Timber.d("User data saved in database");
+                    },
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Indentation is not correct

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 27]
```diff
@@ -0,0 +1,75 @@
...
+                        .getDefaultInstance()
+                        .saveUser(user))
+                .subscribe(() -> {
+                    loadUserFromNetworkResponse.setValue(RETRIEVE_SUCCESSFUL);
+                    Timber.d("User data saved in database");
+                },
+                throwable -> {
+                    loadUserFromNetworkResponse.setValue(RETRIEVE_UNSUCCESSFUL);
+                    Timber.d(throwable.getMessage() + " Error getting data from network");
+                });
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Indentation is still wrong

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java [hunk 28]
```diff
@@ -0,0 +1,75 @@
...
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .flatMapCompletable(user -> RealmDataRepository
+                        .getDefaultInstance()
+                        .saveUser(user))
+                .subscribe(() -> {
+                    loadUserFromNetworkResponse.setValue(RETRIEVE_SUCCESSFUL);
+                    Timber.d("User data saved in database");
+                },
+                throwable -> {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java]: Move throwable to previous line

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Comment by simarsingh24** [comment]
@iamareebjamal should I squash the commits?

**Comment by dr0pdb** [comment]
@simarsingh24 Yes and please update the branch also as it is out-of-date.

**Review by iamareebjamal — APPROVED** [review]
